middle_end/inconstant_idents.ml

```diff
@@ -88,43 +89,61 @@ module NotConstants(P:Param) = struct
   let symbol_field_dep_table : dep list Symbol_field.Tbl.t =
     Symbol_field.Tbl.create 100
 
+  (* adds 'dep in NC' *)
+  let mark_dep =function
+    | Var id ->
+      if not (Variable.Tbl.mem variables id)
+      then Variable.Tbl.add variables id ()
+    | Closure cl ->
+      if not (Set_of_closures_id.Tbl.mem closures cl)
+      then Set_of_closures_id.Tbl.add closures cl ()
+    | Symbol i ->
+      if not (Symbol.Tbl.mem symbols i)
+      then Symbol.Tbl.add symbols i ()
+    | Symbol_field i ->
+      if not (Symbol_field.Tbl.mem symbol_fields i)
+      then Symbol_field.Tbl.add symbol_fields i ()
+
+  (* adds 'curr in NC' *)
+  let mark_curr curr =
```

Comment:
We could probably improve that quite a lot by adding (the future annotation) [@unrolled] at the right call sites.
Or more directly mark `mark_dep` as [@inline] and call it instead of `mark_curr` in some situations.